tests/mocks/MockServer.hh

```diff
@@ -28,11 +29,13 @@ public:
     template<typename TApi>
     void HandleRequest(std::function<bool(const typename TApi::Request&, typename TApi::Response&)>&& fHandler)
     {
-        m_mHandlers.insert_or_assign(std::string(TApi::Name()), [fHandler = std::move(fHandler)](
-                                                                    void* restrict pRequest, void* restrict pResponse) {
-            const gsl::not_null<const typename TApi::Request* const> pTRequest{
+        m_mHandlers.insert_or_assign(
+            std::string(TApi::Name()),
+            [fHandler = std::move(fHandler)](const void* restrict pRequest, void* restrict pResponse) 
```

Comment:
pRequest needs to be const void* to not use const_cast as well as in the map. It makes sense since all requests are passed by const reference and responses by reference.
References and pointers have the same exact semantics just different syntax.